src/lib/libmints/deriv.cc

```diff
@@ -747,14 +751,16 @@ SharedMatrix Deriv::compute()
     corr->add(x_contr_);
     corr->add(tpdm_contr_);
     if (reference_separate) {
-        gradient_->add(x_ref_contr_);
-        gradient_->add(opdm_ref_contr_);
-        gradient_->add(tpdm_ref_contr_);
-        SharedMatrix scf_gradient(gradient_->clone());
-        scf_gradient->set_name("Reference Gradient");
-        scf_gradient->print_atom_vector();
-        wfn_->reference_wavefunction()->set_gradient(scf_gradient);
-        corr->print_atom_vector();
+        if ( x_ref_contr_ && opdm_ref_contr_ && tpdm_ref_contr_) {
```

Comment:
I just ran `git blame deriv.cc` and it looks like I am indeed the guilty party.  I felt exactly the same way as @lothian did when he was [writing the ccsort code](https://github.com/psi4/psi4/blob/master/src/bin/ccsort/classify.cc#L45).